### ISSUE TITLE
Follow-up from "Resolve "convert stellarator.f90 to Python""

### ISSUE THREAD

**Issue by jonmaddock** [ukaea/PROCESS#1841]
In GitLab by @timothy-nunn on Apr 26, 2023, 13:55

The following discussion from !847 should be addressed:

- [ ] @jonmaddock started a [discussion](https://git.ccfe.ac.uk/process/process/-/merge_requests/847#note_239687):  (+3 comments)

    > I don' think this is a good idea: the model running differently because it's aware of which constraints are being applied to it. I believe the returned values will be zero in the case where the constraint isn't active, which I find confusing, and this isn't done anywhere else to my knowledge. If it's purely performance reasoning, then I think this should probably be removed.

**Comment by jonmaddock** [comment]
Has this been dealt with already @timothy-nunn ?

**Comment by timothy-nunn** [comment]
@jonmaddock no. https://github.com/ukaea/PROCESS/blob/main/process/stellarator.py#L161

**Comment by timothy-nunn** [comment]
I will say this issue was annoying when we first encountered it. `power_at_ignition_point` had some side effects that changed the results for a Stellarator that did not enforce icc 91, hence why myself and JL opted to leave this logic alone.

**Comment by timothy-nunn** [comment]
Removing this condition check changes the `stellarator_helias` test:

```
WARNING  test_process_input_files:test_process_input_files.py:133 Variable                     Ref             New        % Change
WARNING  test_process_input_files:test_process_input_files.py:136 ------------------------------------------------------------
WARNING  test_process_input_files:test_process_input_files.py:138 res_eq_con016           2.09e-06       -4.31e-06         -306.65
WARNING  test_process_input_files:test_process_input_files.py:138 eq_con016              -2.09e-09        4.31e-09          306.65
WARNING  test_process_input_files:test_process_input_files.py:138 ineq_con083             1.85e-09       -2.88e-09         -255.51
WARNING  test_process_input_files:test_process_input_files.py:138 res_eq_con002           3.77e-11        1.26e-10          233.37
WARNING  test_process_input_files:test_process_input_files.py:138 eq_con002               1.64e-10        5.45e-10          233.36
WARNING  test_process_input_files:test_process_input_files.py:138 ineq_con024              5.7e-07        1.76e-06          208.25
WARNING  test_process_input_files:test_process_input_files.py:138 ineq_con035              3.3e-08        8.21e-08          148.94
```

Do we want to remove this condition if it is changing the solution (albeit not by much, since only the constraint tolerances are changing)?

**Comment by jjwalkowiak** [comment]
Well, I have mixed feelings. Technically speaking it works (or at least I hope so). Part of the problem is that current implementation is inefficient (it calls physics module few times, save some results and restores some variables to previous values etc.). 
I can't say that I really trust it, but at the same time getting rid of this completely doesn't sound as a good solution. I would prefer to fix it, but I don't have time for this right now (or in predictable future).
If you want to be strict, you should probably remove it for now and I can restore it later. But maybe some intermediate solution can be used, to place it as non-default option, with remark that you should know what you do when you use this.

**Comment by jjwalkowiak** [comment]
In general I think that this is the way. The problem is that you need to copy-paste more than a little bit of physics module to make it work. If you look into p_plasma_loss_mw calculation, you will see that it depends on other parameters which have to be recalculated first, like fusion reaction rate and radiation. But if you are familiar with the content of the physics module, maybe it would be easier for you - I would have to first analyze what is calculated where and so on.
BTW I think it would be nice to separate p_plasma_loss_mw into another staticmethod, because now it is a bit hidden inside confinement time.
